integrations/repo_search_test.go

```diff
@@ -27,6 +31,23 @@ func resultFilenames(t testing.TB, doc *HTMLDoc) []string {
 func TestSearchRepo(t *testing.T) {
 	prepareTestEnv(t)
 
+	repo, err := models.GetRepositoryByOwnerAndName("user2", "repo1")
+	assert.NoError(t, err)
+
+	models.UpdateRepoIndexer(repo)
+
+	log.Printf("Waiting for indexing\n")
+
+	i := 0
+	for {
+		if (repo.IndexerStatus != nil && len(repo.IndexerStatus.CommitSha) != 0) || i > 60 {
+			break
+		}
+		time.Sleep(1 * time.Second)
+		i++
+	}
+	log.Printf("Indexing took: %d s\n", i)
```

Comment:
What happens if the indexing didn't finish after 60 seconds? Should this check to see if it finished and fail with a message if not?

Reply:
I'm not sure that's much we can really infer much about indexing - the API is very minimal - if anything inferring that the indexing has finished by checking IndexerStatus,CommitSha is non-zero is probably assuming too much. Yes it appears to work and looking at the source code that appears the only way to check but I don't think we can say much more than that.

If after 60s the IndexerStatus.CommitSha hasn't been set then maybe the "API" of indexer has changed? At present this test would just check and hopefully pass - probably taking too long to do so. If however, you enforce that the IndexerStatus has to be set then we're now forcing an API - one which I would not design this way. I guess it depends on how much you want to enforce an API on UpdateRepoIndexer.

Reply:
Sure, that all makes sense and I wouldn't want to make any big changes or block this fix -- just to avoid confusion by saying it has finished if really we just hit the time limit. Maybe just:

```
if i < 60 {
log.Printf("Indexing took: %d s\n", i)
}  else {
log.Printf("Waited the limit: %d s for indexing, continuing\n", i)
}
```
Which keeps the current assumptions from the loop but specifies if we think it finished vs it reached the time limit (since reaching the time limit is probably the only thing we would know for sure here)